docs/audio.rst

```diff
@@ -83,8 +86,215 @@ The built-in sounds can be called using ``audio.play(Sound.NAME)``.
+        ``WAVE_NOISE`` (randomly generated noise).
+    :param fx: Effect to add on the sound, one of the following values:
+        ``FX_TREMOLO``, ``FX_VIBRATO``, ``FX_WARBLE``, or ``None``.
+    :param interpolation: The type of curve between the start and end
+        frequencies, different wave shapes have different rates of change
+        in frequency. One of the following values: ``INTER_LINEAR``,
+        ``INTER_CURVE``, ``INTER_LOG``.
+
+    .. py:attribute:: freq_start
+
+        Start Frequency in Hertz (Hz)
+
+    .. py:attribute:: freq_end
+
+        End Frequency in Hertz (Hz)
+
+    .. py:attribute:: duration
+
+        Duration of the sound (ms), eg: ``500``
+
+    .. py:attribute:: vol_start
+
+        Start volume value, range 0-255, eg: ``120``
+
+    .. py:attribute:: vol_end
+
+        End volume value, range 0-255, eg: ``255``
+
+    .. py:attribute:: wave
+
+        Type of wave shape, one of these values: ``WAVE_SINE``,
+        ``WAVE_SAWTOOTH``, ``WAVE_TRIANGLE``, ``WAVE_SQUARE``,
+        ``WAVE_NOISE`` (randomly generated noise).
+
+    .. py:attribute:: fx
+
+        Effect to add on the sound, one of the following values:
+        ``FX_TREMOLO``, ``FX_VIBRATO``, ``FX_WARBLE``, or ``None``.
+
+    .. py:attribute:: interpolation
+
+        The type of curve between the start and end
+        frequencies, different wave shapes have different rates of change
+        in frequency. One of the following values: ``INTER_LINEAR``,
+        ``INTER_CURVE``, ``INTER_LOG``.
+
+The arguments used to create any Sound Effect, including the built in effects,
+can be inspected by looking at each of the Effect instance attributes, or by
+converting the instance into a string (which can be done via ``str()``
+function, or by using a function that does the conversion automatically like
+``print()``).
+
+For example, with the :doc:`REPL </devguide/repl>` you can inspect the built
+in Effects::
+
+    >>> audio.Effect.CROAK
+    Effect(freq_start=..., freq_end=..., duration=..., vol_start=..., vol_end=..., wave=..., fx=..., interpolation=...)
+
+The built in Effects are immutable, so they cannot be changed. Trying to modify
+a built in Effect will throw an exception::
+
+    >>> audio.Effect.CLICK.duration = 1000
+    Traceback (most recent call last):
+        File "<stdin>", line 1, in <module>
+    TypeError: effect cannot be modified
+
+But a new one can be created like this::
+
+    >>> click_clone = Effect(audio.Effect.CLICK)
+    >>> click_clone.duration = 1000
+    >>>
+
+Built in Sound Effects
+----------------------
+
+Some pre-created Sound Effects are already available as examples. These can
+be played directly ``audio.play(audio.Effect.SQUEAK)``, or used as a base to
+create new effects ``audio.Effect(audio.Effect.SQUEAK, duration=2000)``.
+
+* ``audio.Effect.SQUEAK``
+* ``audio.Effect.WARBLE``
+* ``audio.Effect.CHIRP``
+* ``audio.Effect.CROAK``
+* ``audio.Effect.CLICK``
+
```

Comment:
I was really happy with including these, but I think their relationship to the built-in sounds is curious to explain. I don't know that this means we shouldn't have them, but I'm struggling to work out what to call them. 

`audio.SoundEffect.CROAK` for example is sort of weird to distinguish from `Sound.TWINKLE` - why can one of them be used as a preset and not the other? ("because one is a SoundEffect and one is a Sound"?)

This is one downside to "SoundEffect" over pure "Effect" - it ties the built-in sounds more closely to the SoundEffects - though @microbit-carlos points out that it does make sense that a 'Sound' is made out of SoundEffects...

Reply:
The existing built-in sounds could be rewritten as a (constant) tuple of (constant) `SoundEffect` instances.  This is similar to sequences of images, like all the clock faces.

Eg:
```python
>>> audio.Sound.SPRING
(SoundEffect(...), SoundEffect(...))
```
Then users could really see how `Sound`'s are made and tweak them if they like.

Reply:
Yes, this is something that was quite desirable, our only concern is all the extra flash usage, specially if the strings are already in CODAL and probably cannot be excluded as they are.
Would be good to get an estimation of how much extra flash space this would consume?